assets/javascripts/modules/checkout/validations.js

```diff
@@ -46,13 +46,27 @@ define([
         return !validity.emptyFields.length && validity.hasConfirmedEmail && validity.hasValidEmail;
     }
 
+    function displayMandatoryValidityErrors(validity) {
+        renderEmailError(!validity.hasValidEmail);
+        toggleError(form.$EMAIL_CONTAINER, !validity.hasValidEmail);
+        toggleError(form.$CONFIRM_EMAIL_CONTAINER, validity.hasValidEmail && !validity.hasConfirmedEmail);
+    }
+
+
+    /**
+     * TODO:
+     * Rather than querying DOM here, this should take
```

Comment:
Done: https://trello.com/c/G9zUsL30/89-improve-form-validation-logic Thinks it's worth keeping the comments here too though.